hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/acl/SharedTmpDirAuthorizer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.security.acl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+
+import java.util.Objects;
+
+/**
+ * SharedTmp implementation of {@link IAccessAuthorizer}.
+ */
+public class SharedTmpDirAuthorizer implements IAccessAuthorizer {
+
+  private final OzoneNativeAuthorizer ozoneNativeAuthorizer;
+  private final IAccessAuthorizer authorizer;
+
+  public SharedTmpDirAuthorizer(OzoneNativeAuthorizer ozoneNativeAuthorizer,
+      IAccessAuthorizer authorizer) {
+    this.ozoneNativeAuthorizer = ozoneNativeAuthorizer;
+    this.authorizer = authorizer;
+  }
+
+  /**
+   * Check access for a given ozoneObject.
+   *
+   * @param ozObject object for which access needs to be checked.
+   * @param context Context object encapsulating all user related information.
+   * @return true if user has access else false.
+   */
+  @Override
+  public boolean checkAccess(IOzoneObj ozObject, RequestContext context)
+      throws OMException {
+    Objects.requireNonNull(ozObject);
+    Objects.requireNonNull(context);
+    OzoneObjInfo objInfo;
+
+    if (ozObject instanceof OzoneObjInfo) {
+      objInfo = (OzoneObjInfo) ozObject;
+      if (StringUtils.isNotEmpty(objInfo.getVolumeName()) &&
+          StringUtils.isNotEmpty(objInfo.getBucketName()) &&
+          objInfo.getVolumeName().equals("tmp") &&
+          objInfo.getBucketName().equals("tmp")) {
```

Comment:
Let's extract this condition to a method in `OFSPath` and use the constants in that class.

Reply:
Updated the PR with this.

Reply:
@adoroszlai Thanks for updating the PR. Sorry, for not responding, I was on vacation.